api/services/sic_lookup_client.py

```diff
@@ -28,11 +29,16 @@ def __init__(self, data_path: str | None = None) -> None:
         """Initialise the SIC lookup client.
 
         Args:
-            data_path: Path to the SIC data file. If not provided, the default
-                knowledge base path will be used.
+            data_path: Path to the SIC data file for initialisation-time configuration.
+                If not provided, the default knowledge base path will be used.
         """
-        # Use the provided path or default path
-        resolved_path = self._get_default_path() if data_path is None else data_path
+        # Always call _get_default_path() to get the resolved path
+        resolved_path = self._get_default_path()
+
+        # Use the provided data_path if it's passed in, otherwise use the result
+        # from _get_default_path()
+        if data_path is not None:
```

Comment:
36 and 40 can be combined to the original code.
resolved_path = self._get_default_path() if data_path is None else data_path